### ISSUE TITLE
Metadata fields not linking to entry fields

### ISSUE THREAD

**Issue by vsteenhoudt** [barrelstrength/craft-sprout-seo#194]
### Description (screenshots, logs, ...)
I've created a new Metadata (sprout seo) field and assigned the Meta Title and description to existing fields, but it doesn't seem they are linked.
When I edit or create an entry the SEO tag is not visible next to the fields. The output in the meta tags in the html remains the global metadata and not the data specified in the entry.

### Additional info

- Craft version:  3.4.9
- PHP version: 7.2.15
- Database driver & version:  MySQL 5.7.26
- Plugins & versions:
Asset Rev   6.0.2
Control Panel CSS   2.2.0
Dashboard Begone    1.0.1
Element API 2.6.0
Enupal Translate    1.2.0
Imager  v2.4.0
Mailgun 1.4.2
Navigation  1.3.6
Redactor    2.6.0.1
Sprout Fields   3.6.1
Sprout SEO  4.3.3
Store Hours 2.1.1.1
Tags    1.0.6
Typed link field    1.0.20
Video Embedder  1.1.4

<img src="https://frontapp.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_1fqf7)

**Comment by BenParizek** [comment]
Hi @vsteenhoudt 

I've just released Sprout SEO 4.4.1 which addresses a few issues about the way metadata is was being saved in the last release where we added Craft 3.4 support. Can you try to update to the latest and see if this resolves your issues? Let me know if you see anything outstanding after the update to v4.4.1.

**Comment by lexislav** [comment]
Hi,

I have updated to 4.4.1.2 and no changes for me on this. Metadata field (translatable for every web) seems not to be linked. The output in meta tags remains the global metadata. No Meta Title (linked to Title) and Description (linked to perex redactor field) are rendered.

However when I turn on the OG details and fill those out, it works as it should and OG title and OG description is used in header metas, page title etc.

Only change when update to 4.4.1.2 that in dev mode I am getting PHP worning:

```
yii\base\ErrorException: mb_strpos() expects parameter 1 to be string, object given in /Users/lexislav/Websites/koma-family.cz/vendor/barrelstrength/sprout-seo/src/base/MetaImageTrait.php:82
Stack trace:
#0 /Users/lexislav/Websites/koma-family.cz/vendor/craftcms/cms/src/web/ErrorHandler.php(74): yii\base\ErrorHandler->handleError(2, 'mb_strpos() exp...', '/Users/lexislav...', 82)
#1 [internal function]: craft\web\ErrorHandler->handleError(2, 'mb_strpos() exp...', '/Users/lexislav...', 82, Array)
#2 /Users/lexislav/Websites/koma-family.cz/vendor/barrelstrength/sprout-seo/src/base/MetaImageTrait.php(82): mb_strpos(Object(barrelstrength\sproutseo\models\Metadata), 'http')
#3 /Users/lexislav/Websites/koma-family.cz/vendor/barrelstrength/sprout-seo/src/meta/OpenGraphMetaType.php(619): barrelstrength\sproutseo\meta\OpenGraphMetaType->prepareAssetMetaData(Object(barrelstrength\sproutseo\models\Metadata), 'sproutSeo-socia...', false)
#4 /Users/lexislav/Websites/koma-family.cz/vendor/barrelstrength/sprout-seo/src/models/Metadata.php(460): barrelstrength\sproutseo\meta\OpenGraphMetaType->getMetaTagData()
#5 /Users/lexislav/Websites/koma-family.cz/vendor/barrelstrength/sprout-seo/src/services/Optimize.php(166): barrelstrength\sproutseo\models\Metadata->getMetaTagData()
#6 /Users/lexislav/Websites/koma-family.cz/vendor/barrelstrength/sprout-seo/src/services/Optimize.php(107): barrelstrength\sproutseo\services\Optimize->getMetadata(Object(craft\models\Site), true, Array)
#7 /Users/lexislav/Websites/koma-family.cz/storage/runtime/compiled_templates/ca/caa1e87f29ed08b23de90b51d957540542e989c33b8c367f47e32d38a2ba185b.php(127): barrelstrength\sproutseo\services\Optimize->getMetadataViaContext(Array)
#8 /Users/lexislav/Websites/koma-family.cz/vendor/twig/twig/src/Template.php(407): __TwigTemplate_d6879c4d7ffa3f0e848532715be76d91f167f3deb0b30701a6c4491a92b00432->doDisplay(Array, Array)
#9 /Users/lexislav/Websites/koma-family.cz/vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#10 /Users/lexislav/Websites/koma-family.cz/storage/runtime/compiled_templates/0f/0f717546cd5cf024e9b50cce32e8b24ac726c6f33274febe912be0770ca29f7e.php(45): Twig\Template->display(Array, Array)
#11 /Users/lexislav/Websites/koma-family.cz/vendor/twig/twig/src/Template.php(407): __TwigTemplate_4ca9dd8976705d5f5b2ce3a292370d7c84c24630eb10d54e44bb5bff344c2142->doDisplay(Array, Array)
#12 /Users/lexislav/Websites/koma-family.cz/vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#13 /Users/lexislav/Websites/koma-family.cz/vendor/twig/twig/src/Template.php(392): Twig\Template->display(Array)
#14 /Users/lexislav/Websites/koma-family.cz/vendor/twig/twig/src/TemplateWrapper.php(45): Twig\Template->render(Array, Array)
#15 /Users/lexislav/Websites/koma-family.cz/vendor/twig/twig/src/Environment.php(318): Twig\TemplateWrapper->render(Array)
#16 /Users/lexislav/Websites/koma-family.cz/vendor/craftcms/cms/src/web/View.php(397): Twig\Environment->render('reference/_entr...', Array)
#17 /Users/lexislav/Websites/koma-family.cz/vendor/craftcms/cms/src/web/View.php(458): craft\web\View->renderTemplate('reference/_entr...', Array)
#18 /Users/lexislav/Websites/koma-family.cz/vendor/craftcms/cms/src/web/Controller.php(235): craft\web\View->renderPageTemplate('reference/_entr...', Array, 'site')
#19 /Users/lexislav/Websites/koma-family.cz/vendor/craftcms/cms/src/controllers/TemplatesController.php(98): craft\web\Controller->renderTemplate('reference/_entr...', Array)
#20 [internal function]: craft\controllers\TemplatesController->actionRender('reference/_entr...', Array)
#21 /Users/lexislav/Websites/koma-family.cz/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#22 /Users/lexislav/Websites/koma-family.cz/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#23 /Users/lexislav/Websites/koma-family.cz/vendor/craftcms/cms/src/web/Controller.php(178): yii\base\Controller->runAction('render', Array)
#24 /Users/lexislav/Websites/koma-family.cz/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('render', Array)
#25 /Users/lexislav/Websites/koma-family.cz/vendor/craftcms/cms/src/web/Application.php(291): yii\base\Module->runAction('templates/rende...', Array)
#26 /Users/lexislav/Websites/koma-family.cz/vendor/yiisoft/yii2/web/Application.php(103): craft\web\Application->runAction('templates/rende...', Array)
#27 /Users/lexislav/Websites/koma-family.cz/vendor/craftcms/cms/src/web/Application.php(276): yii\web\Application->handleRequest(Object(craft\web\Request))
#28 /Users/lexislav/Websites/koma-family.cz/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#29 /Users/lexislav/Websites/koma-family.cz/web/index.php(21): yii\base\Application->run()
#30 {main}
```

**Comment by vsteenhoudt** [comment]
Hey @BenParizek ,
I've just updated and looks like I'm getting the same error as lexislav.
I've also noticed a js error in the console window:

`jquery.js:3850 Uncaught TypeError: Cannot read property 'classList' of undefined
    at e.value (sproutseo.js?v=1583312095:2)
    at new e (sproutseo.js?v=1583312095:2)
    at HTMLDocument.<anonymous> (488-test-seo-page:1607)
    at e (jquery.js:3557)
    at t (jquery.js:3625)`

`yii\base\ErrorException: mb_strpos() expects parameter 1 to be string, object given in /var/www/html/vendor/barrelstrength/sprout-seo/src/base/MetaImageTrait.php:82
Stack trace:
#0 /var/www/html/vendor/craftcms/cms/src/web/ErrorHandler.php(74): yii\base\ErrorHandler->handleError(2, 'mb_strpos() exp...', '/var/www/html/v...', 82)
#1 [internal function]: craft\web\ErrorHandler->handleError(2, 'mb_strpos() exp...', '/var/www/html/v...', 82, Array)
#2 /var/www/html/vendor/barrelstrength/sprout-seo/src/base/MetaImageTrait.php(82): mb_strpos(Object(barrelstrength\sproutseo\models\Metadata), 'http')
#3 /var/www/html/vendor/barrelstrength/sprout-seo/src/meta/OpenGraphMetaType.php(619): barrelstrength\sproutseo\meta\OpenGraphMetaType->prepareAssetMetaData(Object(barrelstrength\sproutseo\models\Metadata), 'sproutSeo-ogRec...', false)
#4 /var/www/html/vendor/barrelstrength/sprout-seo/src/models/Metadata.php(460): barrelstrength\sproutseo\meta\OpenGraphMetaType->getMetaTagData()
#5 /var/www/html/vendor/barrelstrength/sprout-seo/src/services/Optimize.php(166): barrelstrength\sproutseo\models\Metadata->getMetaTagData()
#6 /var/www/html/vendor/barrelstrength/sprout-seo/src/services/Optimize.php(107): barrelstrength\sproutseo\services\Optimize->getMetadata(Object(craft\models\Site), true, Array)
#7 /var/www/html/storage/runtime/compiled_templates/21/21f40c5bcd1e5d1b9afd04bc6e6babcc2de7c85033b3316fa8d7a3bf08700989.php(54): barrelstrength\sproutseo\services\Optimize->getMetadataViaContext(Array)
#8 /var/www/html/vendor/twig/twig/src/Template.php(407): __TwigTemplate_cadd4d7108aaf589d7e2e99710c13a70b6149118f80cd7b9e61391c4c20926fb->doDisplay(Array, Array)
#9 /var/www/html/vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#10 /var/www/html/storage/runtime/compiled_templates/24/248bf9b4fbcf0f0b0a7a00b97cfe44509775a29eb38d939ceafd7a9f433ee123.php(51): Twig\Template->display(Array, Array)
#11 /var/www/html/vendor/twig/twig/src/Template.php(407): __TwigTemplate_c86e661989619933655dbd619adcfe2982a1f92ed75de6213856e530fa921cea->doDisplay(Array, Array)
#12 /var/www/html/vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#13 /var/www/html/storage/runtime/compiled_templates/74/74bac9cd15f463de84bfeb1a79889caa7cf6f16e0d38d566c66ba25199df5896.php(42): Twig\Template->display(Array)
#14 /var/www/html/vendor/twig/twig/src/Template.php(407): __TwigTemplate_1d703ab420d3dcae60091608e0d9732d0bd17690e590a9143ff5bd4be755ed27->doDisplay(Array, Array)
#15 /var/www/html/vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#16 /var/www/html/vendor/twig/twig/src/Template.php(392): Twig\Template->display(Array)
#17 /var/www/html/vendor/twig/twig/src/TemplateWrapper.php(45): Twig\Template->render(Array, Array)
#18 /var/www/html/vendor/twig/twig/src/Environment.php(318): Twig\TemplateWrapper->render(Array)
#19 /var/www/html/vendor/craftcms/cms/src/web/View.php(397): Twig\Environment->render('_pages/_entry', Array)
#20 /var/www/html/vendor/craftcms/cms/src/web/View.php(458): craft\web\View->renderTemplate('_pages/_entry', Array)
#21 /var/www/html/vendor/craftcms/cms/src/web/Controller.php(235): craft\web\View->renderPageTemplate('_pages/_entry', Array, 'site')
#22 /var/www/html/vendor/craftcms/cms/src/controllers/TemplatesController.php(98): craft\web\Controller->renderTemplate('_pages/_entry', Array)
#23 [internal function]: craft\controllers\TemplatesController->actionRender('_pages/_entry', Array)
#24 /var/www/html/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#25 /var/www/html/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#26 /var/www/html/vendor/craftcms/cms/src/web/Controller.php(178): yii\base\Controller->runAction('render', Array)
#27 /var/www/html/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('render', Array)
#28 /var/www/html/vendor/craftcms/cms/src/web/Application.php(291): yii\base\Module->runAction('templates/rende...', Array)
#29 /var/www/html/vendor/yiisoft/yii2/web/Application.php(103): craft\web\Application->runAction('templates/rende...', Array)
#30 /var/www/html/vendor/craftcms/cms/src/web/Application.php(276): yii\web\Application->handleRequest(Object(craft\web\Request))
#31 /var/www/html/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#32 /var/www/html/web/index.php(21): yii\base\Application->run()
#33 {main}`

**Comment by BenParizek** [comment]
I can recreate this. Looking into a fix.

**Comment by BenParizek** [comment]
Hi Folks. I believe I've got a fix for this but I'm probably not going to have a chance to test it more rigorously or get a necessary migration written this week.

For now, for any newcomers looking at this thread I'd recommend holding off upgrading until that next release and for anybody on this release please be aware that in the most recent release there is a related bug that may store the incorrect canonical value as the CP entries URL when you save your entries. Until I get the release out, the only fix for that would be to edit and remove that value from your db directly.

I'll update this thread as soon as I get the release out.

**Comment by lexislav** [comment]
Thanks - there is some impovement.
But still geting an error on metadata field (settings: title - title, description - {perex}, keywords editable field) - on frontend.
Hope next release will fix this.

```
yii\base\ErrorException: mb_strpos() expects parameter 1 to be string, object given in /Users/lexislav/Websites/koma-family.cz/vendor/barrelstrength/sprout-seo/src/base/MetaImageTrait.php:82
Stack trace:
#0 /Users/lexislav/Websites/koma-family.cz/vendor/craftcms/cms/src/web/ErrorHandler.php(74): yii\base\ErrorHandler->handleError(2, 'mb_strpos() exp...', '/Users/lexislav...', 82)
#1 [internal function]: craft\web\ErrorHandler->handleError(2, 'mb_strpos() exp...', '/Users/lexislav...', 82, Array)
#2 /Users/lexislav/Websites/koma-family.cz/vendor/barrelstrength/sprout-seo/src/base/MetaImageTrait.php(82): mb_strpos(Object(barrelstrength\sproutseo\models\Metadata), 'http')
#3 /Users/lexislav/Websites/koma-family.cz/vendor/barrelstrength/sprout-seo/src/meta/OpenGraphMetaType.php(619): barrelstrength\sproutseo\meta\OpenGraphMetaType->prepareAssetMetaData(Object(barrelstrength\sproutseo\models\Metadata), 'sproutSeo-socia...', false)
#4 /Users/lexislav/Websites/koma-family.cz/vendor/barrelstrength/sprout-seo/src/models/Metadata.php(242): barrelstrength\sproutseo\meta\OpenGraphMetaType->getMetaTagData()
#5 /Users/lexislav/Websites/koma-family.cz/vendor/barrelstrength/sprout-seo/src/services/Optimize.php(166): barrelstrength\sproutseo\models\Metadata->getMetaTagData()
#6 /Users/lexislav/Websites/koma-family.cz/vendor/barrelstrength/sprout-seo/src/services/Optimize.php(107): barrelstrength\sproutseo\services\Optimize->getMetadata(Object(craft\models\Site), true, Array)
#7 /Users/lexislav/Websites/koma-family.cz/storage/runtime/compiled_templates/ca/caa1e87f29ed08b23de90b51d957540542e989c33b8c367f47e32d38a2ba185b.php(128): barrelstrength\sproutseo\services\Optimize->getMetadataViaContext(Array)
#8 /Users/lexislav/Websites/koma-family.cz/vendor/twig/twig/src/Template.php(407): __TwigTemplate_d6879c4d7ffa3f0e848532715be76d91f167f3deb0b30701a6c4491a92b00432->doDisplay(Array, Array)
#9 /Users/lexislav/Websites/koma-family.cz/vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#10 /Users/lexislav/Websites/koma-family.cz/storage/runtime/compiled_templates/0f/0f717546cd5cf024e9b50cce32e8b24ac726c6f33274febe912be0770ca29f7e.php(45): Twig\Template->display(Array, Array)
#11 /Users/lexislav/Websites/koma-family.cz/vendor/twig/twig/src/Template.php(407): __TwigTemplate_4ca9dd8976705d5f5b2ce3a292370d7c84c24630eb10d54e44bb5bff344c2142->doDisplay(Array, Array)
#12 /Users/lexislav/Websites/koma-family.cz/vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#13 /Users/lexislav/Websites/koma-family.cz/vendor/twig/twig/src/Template.php(392): Twig\Template->display(Array)
#14 /Users/lexislav/Websites/koma-family.cz/vendor/twig/twig/src/TemplateWrapper.php(45): Twig\Template->render(Array, Array)
#15 /Users/lexislav/Websites/koma-family.cz/vendor/twig/twig/src/Environment.php(318): Twig\TemplateWrapper->render(Array)
#16 /Users/lexislav/Websites/koma-family.cz/vendor/craftcms/cms/src/web/View.php(397): Twig\Environment->render('reference/_entr...', Array)
#17 /Users/lexislav/Websites/koma-family.cz/vendor/craftcms/cms/src/web/View.php(458): craft\web\View->renderTemplate('reference/_entr...', Array)
#18 /Users/lexislav/Websites/koma-family.cz/vendor/craftcms/cms/src/web/Controller.php(235): craft\web\View->renderPageTemplate('reference/_entr...', Array, 'site')
#19 /Users/lexislav/Websites/koma-family.cz/vendor/craftcms/cms/src/controllers/TemplatesController.php(98): craft\web\Controller->renderTemplate('reference/_entr...', Array)
#20 [internal function]: craft\controllers\TemplatesController->actionRender('reference/_entr...', Array)
#21 /Users/lexislav/Websites/koma-family.cz/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#22 /Users/lexislav/Websites/koma-family.cz/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#23 /Users/lexislav/Websites/koma-family.cz/vendor/craftcms/cms/src/web/Controller.php(178): yii\base\Controller->runAction('render', Array)
#24 /Users/lexislav/Websites/koma-family.cz/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('render', Array)
#25 /Users/lexislav/Websites/koma-family.cz/vendor/craftcms/cms/src/web/Application.php(291): yii\base\Module->runAction('templates/rende...', Array)
#26 /Users/lexislav/Websites/koma-family.cz/vendor/yiisoft/yii2/web/Application.php(103): craft\web\Application->runAction('templates/rende...', Array)
#27 /Users/lexislav/Websites/koma-family.cz/vendor/craftcms/cms/src/web/Application.php(276): yii\web\Application->handleRequest(Object(craft\web\Request))
#28 /Users/lexislav/Websites/koma-family.cz/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#29 /Users/lexislav/Websites/koma-family.cz/web/index.php(21): yii\base\Application->run()
#30 {main}
```

**Comment by BenParizek** [comment]
@lexislav Can you share more about your global and field-specific image settings? It looks like the error relates to resolving the image and I've been unable to recreate that in any tests so far.

Can you also confirm that you are using the latest release (from earlier today) v4.4.2?

**Comment by lexislav** [comment]
Sure, related settings are:

Entry section - channel

If the error is related to image, I am trying to assign main entry image as Meta image.
The introImageR is limited to one image, upload path limited to one folder…

Other fields:
title - regular channel title
perex - redactor field translatable for every language. I screenshot I am trying to proceed striptags as it is a redactor field. DOn’t know if this is possible.

Sprout SEO Meta filed is set up to translate for every web.

If I can provide more informations… just let me know.

Thanks

**Comment by BenParizek** [comment]
@lexislav Stripping tags should be fine as long as the syntax is correct. The error here seems to be image related so I don't have any reason to believe right now that the description field is misbehaving.

Can you screenshot the settings you are using for your image? Do you have the Element Metadat field mapped to an existing image field? Are you using a custom pattern? Are you using Meta Details overrides via the Meta Details fields settings or your templates?

Also, if you could please confirm the Sprout SEO version number that would be helpful to make sure we're looking at the same plugin code.

**Comment by lexislav** [comment]
> Can you screenshot the settings you are using for your image? Do you have the Element Metadat field mapped to an existing image field?
> 

YES

> Are you using a custom pattern?
> 

META IMAGE has no custom pattern, I have chose the ntroImageR from dropdown
> Are you using Meta Details overrides via the Meta Details fields settings or your templates?
> 
> 

Right now I am not using Meta Details
> Also, if you could please confirm the Sprout SEO version number that would be helpful to make sure we're looking at the same plugin code.
> 
> 

Sprout SEO PRO 4.4.2

And the bug is related to the introImageR fiedl:
when I remove it from Meta field, it works as it should!


Thanks

A*

**Comment by vsteenhoudt** [comment]
Hello @BenParizek,
I've installed the sprout seo 4.4.2 update. 
I've noticed that the meta fields get mapped when creating new content but not when editing existing content from the same entry type. Existing content still uses global seo meta's.
There is also an error when creating new content:
`yii\base\ErrorException: mb_strpos() expects parameter 1 to be string, object given in /var/www/html/vendor/barrelstrength/sprout-seo/src/base/MetaImageTrait.php:82
Stack trace:
#0 /var/www/html/vendor/craftcms/cms/src/web/ErrorHandler.php(74): yii\base\ErrorHandler->handleError(2, 'mb_strpos() exp...', '/var/www/html/v...', 82)
#1 [internal function]: craft\web\ErrorHandler->handleError(2, 'mb_strpos() exp...', '/var/www/html/v...', 82, Array)
#2 /var/www/html/vendor/barrelstrength/sprout-seo/src/base/MetaImageTrait.php(82): mb_strpos(Object(barrelstrength\sproutseo\models\Metadata), 'http')
#3 /var/www/html/vendor/barrelstrength/sprout-seo/src/meta/OpenGraphMetaType.php(619): barrelstrength\sproutseo\meta\OpenGraphMetaType->prepareAssetMetaData(Object(barrelstrength\sproutseo\models\Metadata), 'sproutSeo-ogRec...', false)
#4 /var/www/html/vendor/barrelstrength/sprout-seo/src/models/Metadata.php(242): barrelstrength\sproutseo\meta\OpenGraphMetaType->getMetaTagData()
#5 /var/www/html/vendor/barrelstrength/sprout-seo/src/services/Optimize.php(166): barrelstrength\sproutseo\models\Metadata->getMetaTagData()
#6 /var/www/html/vendor/barrelstrength/sprout-seo/src/services/Optimize.php(107): barrelstrength\sproutseo\services\Optimize->getMetadata(Object(craft\models\Site), true, Array)
#7 /var/www/html/storage/runtime/compiled_templates/21/21f40c5bcd1e5d1b9afd04bc6e6babcc2de7c85033b3316fa8d7a3bf08700989.php(54): barrelstrength\sproutseo\services\Optimize->getMetadataViaContext(Array)
#8 /var/www/html/vendor/twig/twig/src/Template.php(407): __TwigTemplate_cadd4d7108aaf589d7e2e99710c13a70b6149118f80cd7b9e61391c4c20926fb->doDisplay(Array, Array)
#9 /var/www/html/vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#10 /var/www/html/storage/runtime/compiled_templates/24/248bf9b4fbcf0f0b0a7a00b97cfe44509775a29eb38d939ceafd7a9f433ee123.php(51): Twig\Template->display(Array, Array)
#11 /var/www/html/vendor/twig/twig/src/Template.php(407): __TwigTemplate_c86e661989619933655dbd619adcfe2982a1f92ed75de6213856e530fa921cea->doDisplay(Array, Array)
#12 /var/www/html/vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#13 /var/www/html/storage/runtime/compiled_templates/74/74bac9cd15f463de84bfeb1a79889caa7cf6f16e0d38d566c66ba25199df5896.php(42): Twig\Template->display(Array)
#14 /var/www/html/vendor/twig/twig/src/Template.php(407): __TwigTemplate_1d703ab420d3dcae60091608e0d9732d0bd17690e590a9143ff5bd4be755ed27->doDisplay(Array, Array)
#15 /var/www/html/vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#16 /var/www/html/vendor/twig/twig/src/Template.php(392): Twig\Template->display(Array)
#17 /var/www/html/vendor/twig/twig/src/TemplateWrapper.php(45): Twig\Template->render(Array, Array)
#18 /var/www/html/vendor/twig/twig/src/Environment.php(318): Twig\TemplateWrapper->render(Array)
#19 /var/www/html/vendor/craftcms/cms/src/web/View.php(397): Twig\Environment->render('_pages/_entry', Array)
#20 /var/www/html/vendor/craftcms/cms/src/web/View.php(458): craft\web\View->renderTemplate('_pages/_entry', Array)
#21 /var/www/html/vendor/craftcms/cms/src/web/Controller.php(235): craft\web\View->renderPageTemplate('_pages/_entry', Array, 'site')
#22 /var/www/html/vendor/craftcms/cms/src/controllers/TemplatesController.php(98): craft\web\Controller->renderTemplate('_pages/_entry', Array)
#23 [internal function]: craft\controllers\TemplatesController->actionRender('_pages/_entry', Array)
#24 /var/www/html/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#25 /var/www/html/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#26 /var/www/html/vendor/craftcms/cms/src/web/Controller.php(178): yii\base\Controller->runAction('render', Array)
#27 /var/www/html/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('render', Array)
#28 /var/www/html/vendor/craftcms/cms/src/web/Application.php(291): yii\base\Module->runAction('templates/rende...', Array)
#29 /var/www/html/vendor/yiisoft/yii2/web/Application.php(103): craft\web\Application->runAction('templates/rende...', Array)
#30 /var/www/html/vendor/craftcms/cms/src/web/Application.php(276): yii\web\Application->handleRequest(Object(craft\web\Request))
#31 /var/www/html/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#32 /var/www/html/web/index.php(21): yii\base\Application->run()
#33 {main}`

**Comment by BenParizek** [comment]
@vsteenhoudt @lexislav  I've just released v4.4.2 which should resolve the error that was being thrown with the mapped Meta Image fields. (For the error: `mb_strpos() expects parameter 1 to be string, object given`)

This thread is getting a bit long and has a few topics going on it. I believe this resolves the image mapping issue and I'm not sure if the issue of fields not mapping correctly still exists for one or both of you. Right now, in my local tests they appear to map correctly for new and existing entries.

That said, I'm going to close out this ticket so we can consider the one issue resolved. And I've created a new ticket for the question around the fields not mapping properly. If that issue still exists for you, can you [add a comment to this new ticket](https://github.com/barrelstrength/craft-sprout-seo/issues/196) confirming that you are still seeing the issue on Sprout SEO v4.4.3 and provide details about your Metadata field settings and anything else that may help me create your specific scenario (assuming this isn't happening to every scenario right now). If you are able to send your database over to sprout@barrelstrengthdesign.com that could also be helpful for me to diagnose what is going on.
